app/jobs/bulkrax/importer_collection_job.rb

```diff
@@ -0,0 +1,13 @@
+module Bulkrax
+  class ImporterCollectionJob < ApplicationJob
+    queue_as :import
+
+    def perform(*args)
+      entry = Entry.find(args[0])
+      importer = Importer.find(args[1])
+      collection = entry.build
```

Comment:
@orangewolf this was my attempt at ensuring the collection gets created first, by passing back the collection object returned from the factory ... I'm not sure that would be entirely reliable. The other thing is that I rather assumed that only one collection would be being created, but I might be mistaken there.

Reply:
I think there is likely still an issue with this creating multiples of a collection while it is running, but it does solve a lot of the other collection related problems and gives us a good place to _put_ that logic if we find we do need it. Unless there is another reason this code is WIP, let's get it merged and try it out and put some locking code in if we still need to.

Reply:
@orangewolf  I've added some locking code - see comments above; I think it's working as it should so I think we can risk a merge and fuller test.

Reply:
@orangewolf I have added a couple of database columns, and I've changed collection_id on entries to collection_ids (an Array), to support multiple collections.